e2etest/pkg/routes/routes.go

```diff
@@ -25,23 +25,22 @@ func init() {
 
 // For IP returns the list of routes in the given container
 // (or in the current host) to reach the service ip.
-func ForIP(target string, exec executor.Executor, ipFamily string) []net.IP {
+func ForIP(target string, exec executor.Executor) []net.IP {
+	dst := net.ParseIP(target)
+	framework.ExpectNotEqual(dst, nil, "Failed to convert", target, "to ip")
 
+	re := Ipv4Re
 	res, err := exec.Exec("ip", []string{"route", "show", target}...)
-	if ipFamily == "ipv6" {
+	framework.ExpectNoError(err)
+
+	if dst.To4() == nil { // assuming it's an ipv6 address
+		re = Ipv6Re
 		res, err = exec.Exec("ip", []string{"-6", "route", "show", target}...)
 	}
 	framework.ExpectNoError(err)
 
 	routes := make([]net.IP, 0)
 
-	dst := net.ParseIP(target)
-	framework.ExpectNotEqual(dst, nil, "Failed to convert", target, "to ip")
-
-	re := Ipv4Re
-	if dst.To4() == nil { // assuming it's an ipv6 address
-		re = Ipv6Re
-	}
```

Comment:
why we dropped v6 support here

Reply:
We did not. I moved the logic above, based on dst being v4 or v6.

Reply:
See https://github.com/metallb/metallb/pull/1077/files/4d8c2b0f70c844d2836307686a34e9c7cbd87f07#diff-463ff4e14d1a72292475b266c5584dda7ed79f381f0759e8b3d14932dfd617e9R29